src/main/java/teammates/storage/sqlapi/UsersDb.java

```diff
@@ -263,4 +263,19 @@ public List<Student> getAllStudentsForEmail(String email) {
         return HibernateUtil.createQuery(cr).getResultList();
     }
 
+    /**
+     * Gets all instructors associated with a googleId.
+     */
+    public List<Instructor> getInstructorsForGoogleId(String googleId) {
+        assert googleId != null;
+
+        CriteriaBuilder cb = HibernateUtil.getCriteriaBuilder();
+        CriteriaQuery<Instructor> cr = cb.createQuery(Instructor.class);
+        Root<Instructor> instructorRoot = cr.from(Instructor.class);
+        Join<Instructor, Account> accountsJoin = instructorRoot.join("account");
```

Comment:
Everytime we want to do a filter query, we need this very complex chunk of code. It would be great if we can somehow abstract this in the root parent class instead. I actually tried before but could not really make it work, not sure if others know the solution to this